tools/azure-sdk-tools/devtools_testutils/resource_testcase.py

```diff
@@ -41,6 +43,7 @@ def __init__(self, name_prefix='',
             self._need_creation = False
         if self.random_name_enabled:
             self.resource_moniker = self.name_prefix + "rgname"
+        self.set_cache(use_cache, location)
```

Comment:
"location" is actually fake in ARM, a resource group is a concept and doesn't exist physically..

I think what you want is to be sure you create two resource groups in this scenario:
```python
@ResourceGroupPreparer(use_cache=True)
@ResourceGroupPreparer(use_cache=True, parameter_name="rg2")
```

which is here would use the same cache.

The only reason someone would want to change the "parameter_name" would be for multiple RG (since this impacts the name of the variables you inject in the tests, this is actually the ONLY way to have two RGs)

I think it means to me the default cache_key should use the parameter_name too

Reply:
++ Good callout both on location and on parameter_name being the actual differentiator, you hit the nail on the head in both cases WRT my intent.